internal/aiptest/create/user_settable_id.go

```diff
@@ -15,14 +15,12 @@ import (
 var userSettableID = suite.Test{
 	Name: "user settable id",
 	Doc: []string{
-		"If method support user settable IDs, when set the resource should",
-		"be returned with the provided ID.",
+		"The created resource should be returned with the ID supplied by the user.",
 	},
 
 	OnlyIf: suite.OnlyIfs(
 		onlyif.HasMethod(aipreflect.MethodTypeCreate),
 		onlyif.MethodNotLRO(aipreflect.MethodTypeCreate),
-		onlyif.HasUserSettableID,
```

Comment:
won't this generate syntax error though?

Reply:
if the code has no user settable ID, we'll generate invalid code?

Reply:
If it doesn't generate invalid code that references a custom ID field that doesn't exist, it should be allright

Reply:
I've been thinking about this one and while the code does work and correctly throws an assertion error, it might be confusing to read because the code that sets the resource names (which then is asserted later) is never generated. Maybe it is better to just rely on the linter here and if the user has decided to ignore the linter than we simply don't generate the test. WDYT?

Reply:
Yeah I agree, let's stick with the linter